build.assets/buildbox/tplibs.mk

```diff
@@ -0,0 +1,235 @@
+# Third-party libraries needed to build Teleport.
+
+mk_dir := $(dir $(lastword $(MAKEFILE_LIST)))
+include $(mk_dir)/bbcommon.mk
+include $(mk_dir)/cross-compile.mk
+
+# We build these libraries ourself and statically link them into the Teleport
+# binary as we need them build with PIE (Position Independent Executable) mode
+# so as to make use of ASLR (Address Space Layout Randomization). We cannot
+# rely on a host OS/packager to have built them this way.
```

Comment:
To be honest, I don't really know. Since `-buildmode=pie` was added to the build in https://github.com/gravitational/teleport/pull/35543 I needed to modify the build Jakub originally had in his draft PR. Jakubs PR was prior to Reed's one just referenced so it all build fine. Once I rebased it to a later base, the libraries stopped building until I could figure out how to build each with PIE mode. That comment reflects that experience. Maybe the libraries in CentOS 7 are already build with PIE mode.

Reply:
> Why didn't we have to do this when building directly on CentOS 7?

This is not true. Reed rebuild all binaries on CentOS 7 to make them PIE compatible. If we want to build Teleport with PIE then all code must be build as PIC. As I mentioned, we can "reduce the complexity" by removing the libbpf from Teleport as OpenSSL by default builds as PIE and we don't really have any other C library that we need to build.

Reply:
There are a number of other C libraries we need to build: libtirpc, libpam, libudev_zero, libcbor, libfido2, libpcsclite.

But I noticed a difference between PIC and PIE. The libraries were already being build PIC, but that was not enough for PIE mode. When I started on this, I had to figure out how to build the libraries for `-buildmode=pie` as the existing PIC build was giving errors when linking.